File: solutions/6-kyu/sort-the-odd/javascript/solution.js
function sortArray(arr) {
  const odd = [];
  const even = [];
  const zeroPos = [];
  for (let i = 0; i < arr.length; i++) {
    if (arr[i] == 0) {
      zeroPos.push(i);
      even.push(arr[i]);
    } else if (arr[i] % 2 != 0) {
      odd.push(arr[i]);
      even.push(0);
    } else {
      even.push(arr[i]);
    }
  }
  odd.sort((a, b) => a - b);
  let count = 0;
  for (let i = 0; i < even.length; i++) {
    if (!zeroPos.includes(i) && even[i] == 0) {
      even[i] = odd[count]
      count++
    }
  }
  return even
}

File: solutions/6-kyu/sort-the-odd/python/solution.py
def sort_array(arr):
    sorted_odds = sorted(i for i in arr if i % 2 != 0)
    evens = [i if i % 2 == 0 else -1 for i in arr]
    return [i if i != -1 else sorted_odds.pop(0) for i in evens]